server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java

```diff
@@ -301,6 +310,7 @@ private MTDocument getTranslationFromMT(MTDocument request,
             // TODO we only return the backendId from the final batch
             backendIdConfirmation = result.getBackendId();
             startBatch = batchEnd;
+            taskHandle.increaseProgress(next.size());
```

Comment:
What's "next" again? We should give it a more descriptive name. textFlowsInBatch or something?